lib/puppet/parser/resource.rb

```diff
@@ -304,6 +305,17 @@ def add_scope_tags
     end
   end
 
+  def replace_sensitive_data
+    if builtin_type?
+      parameters.each_pair do |name, param|
```

Comment:
Looping over every parameter for every resource might be somewhat costly; instead of doing this we could change `#set_parameter` to unwrap sensitive values on parameter assignment. Thoughts?